code/game/machinery/limbgrower.dm

```diff
@@ -0,0 +1,253 @@
+	var/obj/item/weapon/circuitboard/machine/B = new /obj/item/weapon/circuitboard/machine/limbgrower(null)
+	B.apply_default_parts(src)
+	files = new /datum/research/limbgrower(src)
+
+/obj/item/weapon/circuitboard/machine/limbgrower
+	name = "circuit board (Limb Grower)"
+	build_path = /obj/machinery/limbgrower
+	origin_tech = "programming=2;biotech=2"
+	req_components = list(
+							/obj/item/weapon/stock_parts/manipulator = 1,
+							/obj/item/weapon/reagent_containers/glass/beaker = 2,
+							/obj/item/weapon/stock_parts/console_screen = 1)
+
+/obj/machinery/limbgrower/interact(mob/user)
+	if(!is_operational())
+		return
+
+	var/dat = main_win(user)
+
+	switch(screen)
+		if(LIMBGROWER_MAIN_MENU)
+			dat = main_win(user)
+		if(LIMBGROWER_CATEGORY_MENU)
+			dat = category_win(user,selected_category)
+		if(LIMBGROWER_CHEMICAL_MENU)
+			dat = chemical_win(user)
+
+	var/datum/browser/popup = new(user, "Limb Grower", name, 400, 500)
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/limbgrower/deconstruction()
+	for(var/obj/item/weapon/reagent_containers/glass/G in component_parts)
+		reagents.trans_to(G, G.reagents.maximum_volume)
+
+/obj/machinery/limbgrower/attackby(obj/item/O, mob/user, params)
+	if (busy)
+		user << "<span class=\"alert\">The Limb Grower is busy. Please wait for completion of previous operation.</span>"
+		return
+
+	if(default_deconstruction_screwdriver(user, "limbgrower_panelopen", "limbgrower_idleoff", O))
+		updateUsrDialog()
+		return
+
+	if(exchange_parts(user, O))
+		return
+
+	if(panel_open)
+		if(istype(O, /obj/item/weapon/crowbar))
+			default_deconstruction_crowbar(O)
+			return
+
+	if(user.a_intent == "harm") //so we can hit the machine
+		return ..()
+
+	if(stat)
+		return
+
+	if(O.flags & HOLOGRAM)
+		return
+
+/obj/machinery/limbgrower/Topic(href, href_list)
+	if(..())
+		return
+	if (!busy)
+		if(href_list["menu"])
+			screen = text2num(href_list["menu"])
+
+		if(href_list["category"])
+			selected_category = href_list["category"]
+
+		if(href_list["disposeI"])  //Get rid of a reagent incase you add the wrong one by mistake
+			reagents.del_reagent(href_list["disposeI"])
+
+		if(href_list["make"])
+
+			/////////////////
+			//href protection
+			being_built = files.FindDesignByID(href_list["make"]) //check if it's a valid design
+			if(!being_built)
+				return
+
+
+			var/synth_cost = being_built.reagents["synthflesh"]*prod_coeff
+			var/power = max(2000, synth_cost/5)
+
+			if(reagents.has_reagent("synthflesh", being_built.reagents["synthflesh"]*prod_coeff))
+				busy = 1
+				use_power(power)
+				flick("limbgrower_fill",src)
+				icon_state = "limbgrower_idleon"
+				addtimer(src, "buildItem", prod_coeff)
+
+	else
+		usr << "<span class=\"alert\">The limb grower is busy. Please wait for completion of previous operation.</span>"
+
+	updateUsrDialog()
+	return
+
+/obj/machinery/limbgrower/proc/build_item()
+	reagents.remove_reagent("synthflesh",being_built.reagents["synthflesh"]*prod_coeff)
```

Comment:
it's usually a case of outside parties fiddling, i.e admins removing reagents, but it can happen